### PR TITLE
BUG: `weights` is not working for multiple columns in df.plot.hist

### LINKED ISSUE #33173: Setting histogram weights for multiple columns fails
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np
df = pd.DataFrame(dict(zip(['A', 'B'], np.random.randn(2, 100))))
df.plot.hist(weights=0.1*np.ones(shape=(100, 2)))  # fail
```
#### Problem description
Trying to plot a histogram of a multi-column data frame with weights fails with `ValueError: weights should have the same shape as a.` There is no error for a single column. It doesn't work with:
```python
df.plot.hist(weights=0.1*np.ones(shape=(100,)))  # fail
```
either.

#### Expected Output
A `matplotlib` figure.
pandas_hist_weights_bug_01

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None
pandas           : 1.0.3
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.1.1.post20200323
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : 0.48.0

</details>

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [x] closes #33173 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/plotting/test_frame.py [hunk 1]
```diff
@@ -1682,6 +1682,21 @@ def test_hist_df(self):
...
+        # GH 33173
+        np.random.seed(0)
+        df1 = pd.DataFrame(dict(zip(["A", "B"], np.random.randn(2, 100,))))
+
+        ax1 = _check_plot_works(df1.plot, kind="hist", weights=weights)
+
+        df2 = pd.DataFrame(dict(zip(["A", "B"], 0.1 * np.random.randn(2, 100,))))
+        ax2 = _check_plot_works(df2.plot, kind="hist")
+
+        self._check_data(ax1, ax2)
```
Comment (WillAyd) [inline pandas/tests/plotting/test_frame.py:1698]: Is there a way in the test to check that the weights are appropriately rendered by column?

Comment (charlesdong1991) [reply pandas/tests/plotting/test_frame.py]: thanks for taking a look! @WillAyd 

good point, i slightly changed the test, and i think: in order to check weights to be rendered appropriately, we could use `height` in patches, if for instance, weights for all is `0.1` as in original example, the height of plot should be `0.1` times of original one without weights. 

what do you think of this approach?

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
thanks @charlesdong1991

**Pull request merged**